lib/ruby_smb/smb1/packet/nt_create_andx_response.rb

```diff
@@ -0,0 +1,74 @@
+module RubySMB
+  module SMB1
+    module Packet
+
+      # A SMB1 SMB_COM_NT_CREATE_ANDX Response Packet as defined in
+      # [2.2.4.64.2 Response](https://msdn.microsoft.com/en-us/library/ee441612.aspx) and
+      # [2.2.4.9.2 Server Response Extensions](https://msdn.microsoft.com/en-us/library/cc246334.aspx)
+      class NtCreateAndxResponse < RubySMB::GenericPacket
+
+        # A SMB1 Parameter Block as defined by the {NtCreateAndxResponse}
+        class ParameterBlock < RubySMB::SMB1::ParameterBlock
+          endian :little
+
+          struct :words, onlyif: -> { word_count.nonzero? } do
```

Comment:
I added the #words structure to be able to read the response message when the server returns an error. In this case, word_count is zero and no parameter block data is returned. Without this, BinData breaks when reading such error response.

Reply:
Hrm, i'm not sure how i feel about this. I was just using exception handling an the EmtpyPacket class in SMb1 to deal with this before. This seems like a more convenient solution in some respects, however, it adds another layer of nesting to the structure, and in SMB1 these are already getting pretty nested

Reply:
Yeah, I agree with you, it is getting too much nested. Another option would be to add the `onlyif:` parameter to each line:
```ruby
and_x_block             :andx_block,         onlyif: -> { word_count.nonzero? }
# Constants defined in RubySMB::SMB1::OplockLevels
uint8                   :oplock_level,       onlyif: -> { word_count.nonzero? }, label: 'OpLock Level'
uint16                  :fid,                onlyif: -> { word_count.nonzero? }, label: 'FID'
# Constants defined in RubySMB::Dispositions
uint32                  :create_disposition, onlyif: -> { word_count.nonzero? }, label: 'Create Disposition'
file_time               :create_time,        onlyif: -> { word_count.nonzero? }, label: 'Create Time'
...
```
It is a bit of a hack, but it has the same effect without adding a layer of nesting.

I am also fine just using exception handling and the EmtpyPacket class in the client.

Reply:
I removed the #words and #bytes layer of nesting and also removed the `onlyif:` parameter I had originally added on each field (see 56282a3985b88f36df23c470e5f4de9f6be19c35). This can still be re-added later if you think it is worth it. Addressed in b4456bec8c6df3df4e02f7a07e14401506f2945c.